### ISSUE TITLE
[0.44.0] Unable to install consul multiple time on the same K8S cluster

### ISSUE THREAD

**Issue by Lebvanih** [hashicorp/consul-k8s#1250]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

---

### Overview of the Issue
**Environment**: K8S 1.21
**Chart version**: 0.44.0

We are using multiple installation of consul in our cluster. One small consul per customer to have isolation on resource, ACL and other aspect. We were using consul 1.10.3 until recently with chart version 0.32.1 and had no issues.
We are now willing to change our deployment flow and upgrade the consul version/chart at the same time. However we are facing an issue, and from reading the chart source files, I don't see how to avoid it without disabling the ACL. 
In version 0.42.0 of the chart, I see that the ACL are now managed via the K8S auth method, and this seems to rely on ClusterRole, and ClusterRoleBinding. 
The result is, that the first time we install consul, it works, but the second time, helm will complain that the resources already exist and the installation is failing. Is there a way to avoid this behavior but keep the management of the ACL via consul-k8s? 

The other issue that we have related to this, is that the tenant that install consul are not allowed to create ClusterRole in our cluster, but I see that this ticket is still opened https://github.com/hashicorp/consul-k8s/issues/683 and is covering that issue I guess.
### Reproduction Steps
First installation (I downloaded locally the chart to inspect it, but I've the same issue if I use the one direct via the helm repo)
```
pgr@an000335:~/Downloads/consul-k8s-0.44.0$ helm install -n test-pgr consul charts/consul/ -f /tmp/values.yaml 
NAME: consul
LAST DEPLOYED: Wed Jun  1 15:53:37 2022
NAMESPACE: test-pgr
STATUS: deployed
REVISION: 1
NOTES:
Thank you for installing HashiCorp Consul!

Your release is named consul.

To learn more about the release, run:

  $ helm status consul
  $ helm get all consul

Consul on Kubernetes Documentation:
https://www.consul.io/docs/platform/k8s

Consul on Kubernetes CLI Reference:
https://www.consul.io/docs/k8s/k8s-cli
```
Second installation:
```
pgr@an000335:~/Downloads/consul-k8s-0.44.0$ helm install -n test-pgr-2 consul2 charts/consul/ -f /tmp/values.yaml 
Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "consul-cluster-auth-method" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "consul2": current value is "consul"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "test-pgr-2": current value is "test-pgr"
```
### Logs
No logs outside the helm one previously mentionned.

### Expected behavior
We should be able to install consul multiple time on the cluster as it was possible before, either by the system using a RoleBinding if possible rather than a cluster role + cluster role binding, or by manually creating the rbac ressource for that flow if a cluster role is needed (or any other mean if I am missing something). 
I do remember that in an old version there was an issue with the old acl-init job using a clusterrolebinding, but this was changed to use a rolebinding instead. 

### Environment details
**Environment**: K8S 1.21
**Chart version**: 0.44.0
**values.yaml**:
```
client:
  enabled: false
dns:
  enabled: false
global:
  acls:
    manageSystemACLs: true
  datacenter: consul-test
  logJSON: true
  metrics:
    enableAgentMetrics: true
    enabled: true
  name: consul-cluster
server:
  affinity: |
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchLabels:
              app: {{ template "consul.name" . }}
              release: "{{ .Release.Name }}"
              component: server
          topologyKey: kubernetes.io/hostname
      preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 5
        podAffinityTerm:
          topologyKey: "topology.kubernetes.io/region"
          labelSelector:
            matchLabels:
              app: {{ template "consul.name" . }}
              release: "{{ .Release.Name }}"
              component: server
      - weight: 5
        podAffinityTerm:
          topologyKey: "topology.kubernetes.io/zone"
          labelSelector:
            matchLabels:
              app: {{ template "consul.name" . }}
              release: "{{ .Release.Name }}"
              component: server
  annotations: null
  resources:
    limits:
      cpu: 512m
      memory: 512Mi
    requests:
      cpu: 256m
      memory: 256Mi
  storage: 10Gi
  storageClass: ceph-block
```

**Comment by Lebvanih** [comment]
Hello @david-yu,

Indeed it's not really what we were looking for, I did rollback to 0.41.1 for now.

Two questions, if this need to be raised on the discuss of consul let me know, I will gladly re-open it there :) (and I guess the ticket type here could be changed to question as you mention it's not a bug)

What is the reason of the RBAC needing a ClusterRoleBinding rather than a RoleBinding in the case of the K8S auth method implemented for the ACL in 0.42. Is this because it could happen that a client is on another namespace than the servers (as it seems to be the recommended way of deploying)? The documentation of Consul doesn't justify it either (https://www.consul.io/docs/security/acl/auth-methods/kubernetes#rbac). They only mention an example.

While this is not officially supported, If our use case is purely single namespace, could we fork on our side the chart and switch the ClusterRole/ClusterRoleBinding to the Role/RoleBinding equivalent and still use the feature of ACL via Kubernetes Auth.

We are aware that this wouldn't be supported from your side, and we would have to keep an eye and adjust every release, but we are willing to know if this would work.

I am assuming that to have a more clean solution we would need #683, as we could create the Service Account outside and just feed it, but I guess the last feedback on that one is still up to date.

**Comment by david-yu** [comment]
Hi @Lebvanih I will go ahead and close as this is not something we plan on investigating or supporting at the moment. You are welcome to try modifying the Helm chart as is but that would be at your own discretion.
